indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -110,60 +80,143 @@ protected AbstractBatchIndexTask(
   )
   {
     super(id, groupId, taskResource, dataSource, context);
+    segmentLockHelper = new SegmentLockHelper(dataSource);
   }
 
-  public abstract boolean requireLockInputSegments();
+  /**
+   * Return true if this task can overwrite existing segments.
+   */
+  public abstract boolean requireLockExistingSegments();
 
-  public abstract List<DataSegment> findInputSegments(TaskActionClient taskActionClient, List<Interval> intervals)
+  /**
+   * Find segments to lock in the given intervals.
+   * If this task is intend to overwrite only some segments in those intervals, this method should return only those
+   * segments instead of entire segments in those intervals.
+   */
+  // TODO: remove this
```

Comment:
Can you remove the TODO and link to an issue or comment that describes the future work?